sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/UInt128.java

```diff
@@ -55,4 +57,32 @@ public long getLow() {
     public long getHigh() {
         return high;
     }
+
+    public UInt128 add(int value) {
+        UInt128 add = new UInt128(value, 0);
+        long low = this.low + add.low;
+        long high = this.high + add.high;
+
+        if (low < add.low)
+        {
+            high++;
+        }
+
+        return new UInt128(low, high);
+    }
+
+    public UInt128 xor(UInt128 other) {
+        return new UInt128(this.low ^ other.low, this.high ^ other.high);
+    }
+
+    @Override
+    public int compareTo(UInt128 other) {
+        if (this.equals(other)) {
+            return 0;
+        }
+        if ((this.high < this.high) || ((this.high == other.high) && (this.low < other.low))) {
```

Comment:
I think this is a typo -> 

`this.high < this.high` -> should be -> `this.high < other.high` ??

Reply:
oh, yea, definitely a typo

Reply:
Updated